component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -120,6 +128,14 @@ public static String encrypt(String plainText) throws CryptoException {
      */
     public static String decrypt(String cipherText) throws CryptoException {
 
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        // Get custom key from server configuration.
+        String customKey =
```

Comment:
We should be able to refactor this if we change the kernel as mentioned there. Refer kernel PR for more details